src/completion_matcher.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <string_view.h>
+
+class CompletionMatcher {
+ public:
+  // virtual ~CompletionMatcher() = default; // don't know why but it crashed if I uncomment this like, investigating
```

Comment:
Strange, but this should have a virtual dtor for correct behavior (also, the = default should probably go in a .cc file)

Reply:
The `= default` part can behave differently if moved to the `.cc` file.
https://accu.org/index.php/journals/2379

Reply:
No idea whats happened but now it works fine. Probably some weird stuff on my side